### DISCUSSION TITLE
LOW: Show all checks running on merge queue entry

### DISCUSSION THREAD

**Discussion by purajit** [community/community#66920] [Pull Requests]
The merge queue checks UI only shows the status of the required checks. This is often useless, since required checks are often the last step of a workflow. This means that until the workflow reaches its last step, it's very difficult to find the corresponding run. By the time the workflow reaches this last step and it shows up in the checks, it's already too late to be useful.

Showing all running checks would be useful, or least linking to the parent workflow run of a required check.

**Comment by olekxandr** [comment]
It appears you're having trouble with visibility of the entire CI/CD workflow in your GitHub project due to the way checks are handled. GitHub only shows status of required checks in the merge queue, but there doesn't seem to be an easy way to inspect or navigate to the ongoing workflow.

Unfortunately, as of now, GitHub doesn't provide built-in feature to show all running checks or link to the parent workflow run directly from the merge queue. However, there are couple of ways you can navigate to the ongoing workflow which are quite straightforward:

From the repository main page, click on the "Actions" tab. This will list all the recent runs of all workflows in this repository. From this list, you can find the workflow run you're interested in.

From a pull request page, you can directly navigate to the "Checks" tab. This will show all the checks that are part of your CI/CD pipeline, including those that are currently running, completed, or that failed.

Alternatively, if you want to enhance the visibility of your CI/CD pipeline, you can leverage third-party tools such as CircleCI, Jenkins, or TravisCI. These tools often provide a more detailed overview of your pipeline and can integrate with GitHub.

For future, you may want to watch GitHub's feature updates or request this feature from GitHub. In their public roadmap or community forum, they often share plans about upcoming features, and you can provide input to guide their decision making.

I hope this helps, and feel free to ask if you have any more questions!

**Reply by purajit** [reply]
Reporting for useless ChatGPT-generated shpiel

**Comment by marcodicro-dp** [comment]
This would be indeed very useful. Currently I have to click on a finished check, then to the commit (simulated by the Merge Queue in a rebased branch), and only then I can see all the checks being run for that commit.
